components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -2392,7 +2393,7 @@ public static AuthenticatedUser getAuthenticatedUser(AccessTokenDO accessTokenDO
         if (accessTokenDO != null) {
             authenticatedUser = accessTokenDO.getAuthzUser();
         }
-        if (authenticatedUser != null) {
+        if (!OAuth2ServiceComponentHolder.isIDPIdColumnEnabled() && authenticatedUser != null) {
             authenticatedUser.setFederatedUser(isFederatedUser(authenticatedUser));
         }
```

Comment:
Why we set this if IdPColumn not enabled.
Even when it's enabled when loading back it should be set based on the Idp being stored right

Reply:
Will address this change in the next PR